### PR TITLE
Update doneOrAsyncCb to use an overload instead of an intersection

### LINKED ISSUE #4: Update doneOrAsyncCb to use an overload instead of an intersection
We should not have to type each describe / it / beforeEach / afterEach function when testing.

Update the `doneOrAsyncCb` type to use an intersection instead of an union.

### PR DESCRIPTION
**Pull request by jgrund**
Fixes #4.

We should not have to type each describe / it / beforeEach / afterEach function when testing.

Update the doneOrAsyncCb type to use an intersection instead of an union.

Signed-off-by: Joe Grund <grundjoseph@gmail.com>

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [ignore]
-.*\(lib\|test\|builtin-modules\|binary-extensions\|faker\|spdx\).*\.json
 
 [include]
 
 [libs]
 
 [options]
 experimental.strict_type_args=true
+suppress_comment=\\(.\\|\n\\)*\\$ExpectError
```
Comment (johnsonw) [inline .flowconfig]: Didn't we use $FlowIgnore in other repos?

File: package.json [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
   "publishConfig": {
     "access": "public"
   },
   "pre-commit": [
-    "eslint"
+    "eslint",
+    "flow"
   ],
   "repository": {
     "type": "git",
```
Comment (johnsonw) [inline package.json]: The url needs to be fixed here.

File: test/typing.js [hunk 3]
```diff
@@ -1,19 +1,15 @@
 //@flow
 
 describe('my fake suite', () => {
-  beforeEach(() => {
-
-  });
+  beforeEach(() => {});
```
Comment (johnsonw) [inline test/typing.js]: This can be removed.

File: .flowconfig (hunk 1 above)
Comment (jgrund) [inline .flowconfig]: This has a different purpose. It's purpose is to enable us to assert that some tests will fail.

File: test/typing.js (hunk 3 above)
Comment (jgrund) [inline test/typing.js]: This is not running tests, it's testing the flow types.

File: package.json (hunk 2 above)
Comment (jgrund) [inline package.json]: Looks fine to me.

**Pull request merged**